csrc/root_domain_map.cpp

```diff
@@ -942,50 +948,39 @@ void ComputeAtRootDomainMapBuilder::dispatch(Expr* e) {
   visited_.insert(e);
 }
 
-void ComputeAtRootDomainMapBuilder::mapPointwiseOrReductionOp(Expr* e) {
-  if (e->output(0)->getValType() != ValType::TensorView) {
+void ComputeAtRootDomainMapBuilder::mapPointwiseLikeOp(Expr* expr) {
+  if (expr->output(0)->getValType() != ValType::TensorView) {
     return;
   }
 
   // Broadcast is handled separately, so e should never be BroadcastOp.
-  NVF_ERROR(!e->isA<BroadcastOp>());
-  NVF_ERROR(!e->isA<SqueezeOp>());
+  NVF_ERROR(!expr->isA<BroadcastOp>());
+  NVF_ERROR(!expr->isA<SqueezeOp>());
 
-  NVF_ERROR(!e->outputs().empty());
-  const TensorView* out_tv = e->output(0)->as<TensorView>();
-  const TensorDomain* out_td = out_tv->domain();
-  const auto& out_root = out_td->root();
+  NVF_ERROR(!expr->outputs().empty());
+
+  if (expr->outputs().size() > 1) {
+    NVF_ERROR(
+        expr->isA<WelfordOp>() || expr->isA<GroupedReductionOp>() ||
+            expr->isA<GroupedWelfordOp>(),
+        "Unknown multi-output Expr type ",
+        expr->getOpString(),
+        " is found");
+  }
 
   // Record equalities from output to all the inputs
   // ignores non-concretizable broadcasts
-  for (auto* in_tv : ir_utils::filterByType<TensorView>(e->inputs())) {
-    const TensorDomain* in_td = in_tv->domain();
-    std::vector<IterDomain*> in_root =
-        TensorDomain::noReductions(in_tv->getMaybeRFactorDomain());
-    NVF_ERROR(
-        in_root.size() == out_root.size(),
-        "\nExpression: ",
-        e,
-        "\nInput root domain: ",
-        in_root,
-        "\nOutput root domain: ",
-        out_root);
-    for (const auto it : c10::irange(in_root.size())) {
-      if (e->outputs().size() > 1) {
-        NVF_ERROR(
-            e->isA<WelfordOp>() || e->isA<GroupedReductionOp>() ||
-                e->isA<GroupedWelfordOp>(),
-            "Unknown multi-output Expr type ",
-            e->getOpString(),
-            " is found");
-        for (auto out : e->outputs()) {
-          auto out_tv = out->as<TensorView>();
-          auto out_td = out_tv->domain();
-          auto out_root = out_td->root();
-          setMaybeMapped(in_td, in_root[it], out_td, out_root[it]);
-        }
-      } else {
-        setMaybeMapped(in_td, in_root[it], out_td, out_root[it]);
+  for (auto producer_tv : ir_utils::filterByType<TensorView>(expr->inputs())) {
```

Comment:
Don't remember why this wasn't done this way, but just using `PairwiseRootDomainMap` should work fine and simpler.